components/layout_2020/flow/inline.rs

```diff
@@ -77,6 +81,119 @@ struct LinesBoxes {
 }
 
 impl InlineFormattingContext {
+    // This works on an already-constructed `InlineFormattingContext`,
+    // Which would have to change if/when
+    // `BlockContainer::construct` parallelize their construction.
+    #[allow(unused)]
+    pub(super) fn intrinsic_sizes(&self, layout_context: &LayoutContext) -> IntrinsicSizes {
+        struct Computation {
+            paragraph: IntrinsicSizes,
+            current_line: IntrinsicSizes,
+            current_line_percentages: Percentage,
+        }
+        impl Computation {
+            fn traverse(
+                &mut self,
+                layout_context: &LayoutContext,
+                inline_level_boxes: &[Arc<InlineLevelBox>],
+            ) {
+                for inline_level_box in inline_level_boxes {
+                    match &**inline_level_box {
+                        InlineLevelBox::InlineBox(inline_box) => {
+                            let padding = inline_box.style.padding();
+                            let border = inline_box.style.border_width();
+                            let margin = inline_box.style.margin();
+                            macro_rules! add_length {
+                                ($x: expr) => {{
+                                    let length = $x;
+                                    self.current_line.min_content += length;
+                                    self.current_line.max_content += length;
+                                }};
+                            }
+                            macro_rules! add_lengthpercentage {
+                                ($x: expr) => {{
+                                    add_length!($x.length_component());
+                                    self.current_line_percentages += $x.percentage_component();
+                                }};
+                            }
+                            macro_rules! add {
+                                ($condition: ident, $side: ident) => {
+                                    if inline_box.$condition {
+                                        add_lengthpercentage!(padding.$side);
+                                        add_length!(border.$side);
+                                        margin.$side.non_auto().map(|x| add_lengthpercentage!(x));
```

Comment:
Nit: I don't like using `map` if you don't care about the return value and this is only done for side-effects, `if let Some(lp)` would be clearer IMO.

Reply:
Done.